pingrid/impl.py

```diff
@@ -90,6 +79,89 @@ def error_fig(error_msg="error"):
 FuncInterp2d = Callable[[Iterable[np.ndarray]], np.ndarray]
 
 
+class ColorScale:
+    
+    def __init__(self, name, colors, scale=None):
+        self.name = name
+        self.colors = colors
+        if scale is None:
+            self.scale = list(np.arange(len(colors)))
+        else:
+           if (np.diff(scale) < 0).any():
+              raise Exception("scale must be monotically increasing")
+           elif len(colors) == len(scale):
+               self.scale = scale
+           else:
+               raise Exception("scale must be same length as colors")
+
+    def reversed(self, name=None):
+        if name is None:
+            name = self.name + "_r"
+        return ColorScale(name, self.colors[::-1], self.scale)
+
+    def rescaled(self, new_min, new_max):
+        cs_val = np.array(self.scale)
+        scale = (cs_val - cs_val[0]) * (new_max - new_min) / (cs_val[-1] - cs_val[0]) + new_min
+        return ColorScale(self.name, self.colors, scale)
+
+    def to_rgba_array(self, lutsize=256):
+        cs = self.rescaled(0, lutsize-1)
+        nc =  len(cs.scale)
+        colors = np.array(cs.colors)
+        # append output is not used but saves writing a condition dedicated to last color
+        delta_colors  = np.diff(colors, axis=0, append=np.expand_dims(colors[-1,:], 0))
+        delta_scale = np.diff(cs.scale, append=cs.scale[-1])
+        # Construct lutsize x 4 RGBA array
+        rgbaa = np.transpose(np.array([np.piecewise(
+            # Rescale lut indices to colors piece by piece
+            np.arange(lutsize),
+            # Rescale differently from one anchor point to the next
+            [np.arange(lutsize) >= cs.scale[i] for i in range(nc)],
```

Comment:
Don't we need to compare to both lower and upper bounds to avoid having multiple true conditions for a single interval?

Reply:
If we don't want multiple Trues, yes. Right now I am relying on the fact that the last True is the one that counts. Shall I make sure we have unique Trues?

Reply:
I think so, both for efficiency and to avoid relying on undocumented behavior.